wp-content/civi-extensions/civiglific/api/v3/Civiglific/CivicrmGlificContactSyncCron.php

```diff
@@ -0,0 +1,236 @@
+<?php
+
+/**
+ * @file
+ */
+
+use Civi\Api4\Contact;
+use Civi\Api4\SubscriptionHistory;
+use Civi\Api4\Phone;
+use Civi\Api4\GroupContact;
+use Civi\Api4\GlificGroupMap;
+use CRM\Civiglific\GlificClient;
+
+/**
+ * Define API spec.
+ */
+function _civicrm_api3_civiglific_civicrm_glific_contact_sync_cron_spec(&$spec) {}
+
+/**
+ * Cron job to sync contacts between CiviCRM group and Glific group.
+ */
+function civicrm_api3_civiglific_civicrm_glific_contact_sync_cron($params) {
+  $glific = new GlificClient();
+  $returnValues = [];
+
+  $batchSize = 50;
+  // Sleep time between batches (seconds) to reduce API rate limits and server load.
+  $sleepSeconds = 1;
+
+  $glificGroupMaps = GlificGroupMap::get(TRUE)
+    ->addSelect('id', 'group_id', 'collection_id', 'last_sync_date')
+    ->execute()
+    ->jsonSerialize();
+
+  foreach ($glificGroupMaps as $rule) {
+    $civiGroupId = $rule['group_id'];
+    $glificGroupId = $rule['collection_id'];
+    $mapId = $rule['id'];
+    $lastSyncDate = $rule['last_sync_date'];
+
+    if (empty($lastSyncDate)) {
+      $civiContacts = _getCiviContactsFromGroup($civiGroupId);
+    }
+    else {
+      $civiContacts = _getNewlyAddedCiviContacts($civiGroupId, $lastSyncDate);
+    }
+
+    $removedContacts = _getRemovedCiviContacts($civiGroupId, $lastSyncDate);
+
+    if (!empty($removedContacts)) {
+      foreach ($removedContacts as $contact) {
+        try {
+          $normalizedPhone = $contact['phone'];
+          $glificId = $glific->getContactIdByPhone($normalizedPhone);
+
+          if ($glificId) {
+            $glific->removeFromGroup($glificId, $glificGroupId);
+            \Civi::log()->info(
+              "Removed contact {$normalizedPhone} (Glific ID {$glificId}) from Glific group {$glificGroupId}"
+            );
```

Comment:
Consider structured logs over plain strings. They are easier to browse, search.
```suggestion
            \Civi::log()->info(
              'Contact removed from Glific', ['phone' => $normalizedPhone, 'glificId' => $glificId, 'groupId' => $glificGroupId]);
```